ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -137,9 +141,12 @@ def _generate_all_results() -> AllStepResult:
         return all_step_result
 
     try:
-        env = env_factory(
-            worker_id, [env_parameters, engine_configuration_channel, stats_channel]
-        )
+        side_channels = [env_parameters, engine_configuration_channel, stats_channel]
+        if worker_id == 0:
```

Comment:
Need some extra filters, like checking capabilities, and only worker_id==0 should ever send.

Reply:
worker_id == 0 doesn't matter now, but if we extend this to runtime analytics from the built player, we don't want each exe to send events.